google/cloud/bigtable/CMakeLists.txt

```diff
@@ -422,48 +422,72 @@ if (GOOGLE_CLOUD_CPP_ENABLE_CXX_EXCEPTIONS)
-# GNUInstallDirs
-install(TARGETS bigtable_protos bigtable_common_options
-        EXPORT bigtable-targets
-        RUNTIME DESTINATION bin
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
-
-# Install proto generated files into include/google.
-google_cloud_cpp_install_headers(bigtable_protos include)
-
-# Export the CMake targets to make it easy to create configuration files.
-install(EXPORT bigtable-targets
-        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/bigtable_client")
-
-install(TARGETS bigtable_client
-        EXPORT bigtable-targets
-        RUNTIME DESTINATION bin
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
-google_cloud_cpp_install_headers(bigtable_client include/google/cloud/bigtable)
-
-# Setup global variables used in the following *.in files.
-set(GOOGLE_CLOUD_CPP_CONFIG_VERSION_MAJOR ${BIGTABLE_CLIENT_VERSION_MAJOR})
-set(GOOGLE_CLOUD_CPP_CONFIG_VERSION_MINOR ${BIGTABLE_CLIENT_VERSION_MINOR})
-set(GOOGLE_CLOUD_CPP_CONFIG_VERSION_PATCH ${BIGTABLE_CLIENT_VERSION_PATCH})
-set(GOOGLE_CLOUD_CPP_PC_NAME "The Google Cloud Bigtable C++ Client Library")
-set(GOOGLE_CLOUD_CPP_PC_DESCRIPTION
-    "Provides C++ APIs to access Google Cloud Bigtable.")
-set(GOOGLE_CLOUD_CPP_PC_LIBS "-lbigtable_client -lbigtable_protos")
-
-# Create and install the pkg-config files.
-configure_file("${PROJECT_SOURCE_DIR}/google/cloud/config.pc.in"
-               "bigtable_client.pc" @ONLY)
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/bigtable_client.pc" DESTINATION
-              "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
-
-# Create and install the CMake configuration files.
-configure_file("config.cmake.in" "bigtable_client-config.cmake" @ONLY)
-configure_file("${PROJECT_SOURCE_DIR}/google/cloud/config-version.cmake.in"
-               "bigtable_client-config-version.cmake" @ONLY)
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/bigtable_client-config.cmake"
+if ("${GOOGLE_CLOUD_CPP_GRPC_PROVIDER}" STREQUAL "external"
+    OR "${GOOGLE_CLOUD_CPP_PROTOBUF_PROVIDER}" STREQUAL "external")
+    set(GOOGLE_CLOUD_CPP_BIGTABLE_ENABLE_INSTALL "ON")
+else()
+    set(GOOGLE_CLOUD_CPP_BIGTABLE_ENABLE_INSTALL "OFF")
+    message(WARNING [===[
+The install target has been disabled for the bigtable_client library because
+one of the dependencies is set to use external projects. When building against
+external projects the resulting libraries are not installable. The necessary
+dependencies may not installed in your system, or the versions installed in your
+system may not match the version used to compile the google-cloud-cpp libraries.
+Please consult the README.md file for details on how to compile against
+pre-installed versions of the google-cloud-cpp dependencies.
+
+GPRC     PROVIDER = ${GOOGLE_CLOUD_CPP_GRPC_PROVIDER}
+PROTOBUF PROVIDER = ${GOOGLE_CLOUD_CPP_PROTOBUF_PROVIDER}
+
+]===])
+endif ()
+
+if ($"{GOOGLE_CLOUD_CPP_BIGTABLE_DISABLE_INSTALL}")
```

Comment:
Is this meant to be `GOOGLE_CLOUD_CPP_BIGTABLE_ENABLE_INSTALL`?

Reply:
Yes, and it had the `$` and `"` backwards at the beginning.